lib/ClangImporter/ImportDecl.cpp

```diff
@@ -874,6 +874,41 @@ static bool areRecordFieldsComplete(const clang::CXXRecordDecl *decl) {
   return true;
 }
 
+/// Map the access specifier of a Clang record member to a Swift access level.
+///
+/// This mapping is conservative: the resulting Swift access should be at _most_
+/// as permissive as the input C++ access.
+static AccessLevel convertClangAccess(clang::AccessSpecifier access) {
+  switch (access) {
+  case clang::AS_public:
+    // C++ 'public' is actually closer to Swift 'open' than Swift 'public',
+    // since C++ 'public' does not prevent users from subclassing a type or
+    // overriding a method. However, both operations are currently unsupported
+    // across the interop boundary, and it is not clear that Swift 'open'
+    // correctly models other C++ restrictions (e.g., 'final'), so we
+    // conservatively map C++ public to Swift public.
```

Comment:
Swift also has `final`, but as you note, we don't support subclassing a C++ class from Swift, so it's not relevant now.